addon/components/structure-plugin/_private/control-card.gts

```diff
@@ -0,0 +1,195 @@
+import { transactionCombinator } from '@lblod/ember-rdfa-editor/utils/transaction-utils';
+interface Sig {
+  Args: { controller: SayController };
+}
+export default class StructureControlCardComponent extends Component<Sig> {
+  get controller(): SayController {
+    return this.args.controller;
+  }
+  get schema(): Schema {
+    return this.controller.activeEditorState.schema;
+  }
+  get structureType(): NodeType | undefined {
+    return this.schema.nodes['structure'];
+  }
+  get structure(): Option<NodeWithPos> {
+    const currentSelection = this.controller.mainEditorState.selection;
+    if (this.structureType) {
+      return findAncestorOfType(currentSelection, this.structureType) ?? null;
+    }
+    return null;
+  }
+  get structureName(): string {
+    return this.structure?.node.attrs.structureName ?? 'Unknown structure';
+  }
+  get canMoveUp() {
+    return this.controller.checkCommand(moveStructure('up'));
+  }
+  get canMoveDown() {
+    return this.controller.checkCommand(moveStructure('down'));
+  }
+  @action
+  moveStructure(direction: 'down' | 'up') {
+    this.controller.doCommand(moveStructure(direction));
+    this.controller.focus();
+  }
+
+  get canRemoveStructure() {
+    return !!this.structure;
+  }
+  @action
+  removeStructure(withContent: boolean) {
+    if (this.structure) {
+      const { pos, node } = this.structure;
+      if (withContent) {
+        this.controller.withTransaction((tr, state) => {
+          tr.replace(pos, pos + node.nodeSize);
+          return transactionCombinator<boolean>(state, tr)([recalculateNumbers])
+            .transaction;
+        });
+      } else {
+        this.controller.withTransaction((tr, state) => {
+          tr.replaceWith(pos, pos + node.nodeSize, node.content);
+          return transactionCombinator<boolean>(state, tr)([recalculateNumbers])
+            .transaction;
+        });
+      }
+    }
+  }
+  <template>
+    {{#if this.structure}}
+      <AuCard
+        @flex={{true}}
+        @divided={{true}}
+        @expandable={{true}}
+        @shadow={{true}}
+        @isOpenInitially={{true}}
+        @size='small'
+        as |Card|
+      >
+        <Card.header>
+
+          <AuHeading @level='3' @skin='6'>{{t
+              'article-structure-plugin.title.structure-card'
+            }}</AuHeading>
+
+        </Card.header>
+        <Card.content>
+
+          <AuList
+            @divider={{true}}
+            class='au-u-padding-top-tiny au-u-padding-bottom-tiny'
+            as |Item|
+          >
+            <Item>
+              <AuButton
+                @icon={{ChevronUpIcon}}
+                @iconAlignment='left'
+                @skin='link'
+                @disabled={{not this.canMoveUp}}
+                {{on 'click' (fn this.moveStructure 'up')}}
+              >
+                {{t
+                  'structure-plugin.move-up'
+                  structureName=this.structureName
```

Comment:
Yeah I know... on the other hand it's sort of more correct in a way? The document controls its content in the language it uses, and these buttons simply tell you what content they will move (though the down button is clearly wrong ofcourse, forgot that one)

Reply:
Ultimately I think these are better reworked as up/down arrow buttons on the structure itself, but it'll do for now I think

Reply:
I also specifically wanted to avoid the ugliness of passing in the translation function everywhere like we were doing with the article-structures before